dvc/remote/ssh/connection.py

```diff
@@ -9,6 +9,8 @@
 
 try:
     import paramiko
+
+    logging.getLogger("paramiko.transport").disabled = True
```

Comment:
Alternatively, we could explicitly suppress this only for `open_max_sftp_channels()`. But in the event that any SSH/SFTP errors occur, they would be caught and raised/logged as DvcExceptions, so we would just end up with duplicate DVC + paramiko error messages.

Reply:
This only disables `paramiko.transport` logger, right? Should we disable all of the paramiko loggers, or is this enough?

Reply:
I'm not sure exactly. The [SFTPClient](https://github.com/paramiko/paramiko/blob/dccc563c3a581a14ceaa261b5fe0d08539f84b87/paramiko/sftp_client.py#L90) class used in DVC uses the `paramiko.transport` logger. Looking through their code, we could potentially also suppress `paramiko`, `paramiko.transport.sftp`, `paramiko.sftp`, but only using `paramiko.transport` suppresses the specific error message we want to hide.

It's interesting to note that calling `disable_other_loggers()` here still does not suppress the error message. I guess that unless we explicitly reference a name, the loggers aren't actually registered until we instantiate some paramiko client object?

Reply:
@pmrowla Maybe it is because of `paramiko` being loaded dynamically, not sure.

Reply:
>  I guess that unless we explicitly reference a name, the loggers aren't actually registered until we instantiate some paramiko client object?

@pmrowla, @efiop, both are correct. When the `SFTPClient` is instantiated, [it creates a logger](https://github.com/paramiko/paramiko/blob/dccc563c3a581a14ceaa261b5fe0d08539f84b87/paramiko/sftp_client.py#L125-L127). The default is `paramiko.transport` + `.sftp` but anyone can set it via `transport.set_log_channel()` it seems.

And, you can see in the [utils](https://github.com/paramiko/paramiko/blob/dccc563c3a581a14ceaa261b5fe0d08539f84b87/paramiko/util.py#L270-L273), you can see that it instantiates logger from the name.

Reply:
for the record: agreed that we need to put a comment explaining this.

Reply:
Another approach can be setting logger with custom name that we can control and disable it explicitly:
```
self._ssh = paramiko.SSHClient()
self._ssh.set_log_channel("<some_name>)
```

The advantage of this is it won't affect users that are using `paramiko` in the python scripts + dvc `Repo` API (rare combination though).

Reply:
Updated to use `self._ssh.set_log_channel("dvc.paramiko")`, and then explicitly disable that plus `dvc.paramiko.sftp`.